lib/transaction.js

```diff
@@ -251,6 +253,15 @@ class Transaction extends EventEmitter {
         }
       });
   }
+
+  then(onResolve, onReject) {
+    return this._promise.then(onResolve, onReject);
+  }
```

Comment:
FYI: this is probably related to the "`availablePromiseMethods()` list is significantly shorter than the list that it's replacing" concern that I mentioned elsewhere.  The `availablePromiseMethods()` method is currently omitting `"then"`.

In other words: if the list produced by `availablePromiseMethods()` included `"then"`, then you probably could just rely upon meta-programming to generate this method automatically.

(On the flipside: given that the list of Promise methods only contains 2-3 elements now, it might no longer be worth leveraging meta-programming for this.)

Reply:
This will not work, because interface has non-trivial then (but this can be replaced to some internal method too, but this too complex

Reply:
helper which generates array of promise methods helps with optional .finally

Reply:
Right -- we're saying the same thing.  Basically: you can't use meta-programming for this specific case.  However, given that there are only a handful of cases now, it might be worth discarding the meta-programming in favor of something more explicit.

Reply:
done: 9ef63698542dc7e9606d89b3d0b5055c9f47ba61
Sorry for delay, didn't quickly realize your proposed solution